seismicpro/stacking_velocity/velocity_model.py

```diff
@@ -1,201 +1,208 @@
-
-    start_velocity_range = np.array(start_velocity_range, dtype=np.float32)
-    end_velocity_range = np.array(end_velocity_range, dtype=np.float32)
-
-    # Calculate maximal velocity growth (in samples) between two adjacent timestamps,
-    # for which graph nodes are created
-    total_time = (times[-1] - times[0]) / 1000  # from ms to s
-    if max_acceleration is None:
-        max_acceleration = 2 * (np.mean(end_velocity_range) - np.mean(start_velocity_range)) / total_time
-    max_vel_step = np.ceil((max_acceleration * total_time / n_times) / np.mean(velocities[1:] - velocities[:-1]))
-    max_vel_step = np.int32(max_vel_step)
-
-    # Create a graph and find paths with maximal velocity spectrum sum along them to all reachable nodes
-    edges, start_node, end_nodes = create_edges(velocity_spectrum, times, velocities, start_velocity_range,
-                                                end_velocity_range, max_vel_step, n_times, n_velocities)
-    graph = nx.DiGraph()
-    graph.add_weighted_edges_from(zip(*edges))
-    paths = nx.shortest_path(graph, source=start_node, weight="weight")  # pylint: disable=unexpected-keyword-arg
-
-    # Select only paths to the nodes at the last timestamp and choose the optimal one
-    path_weights = [(paths[end_node], nx.path_weight(graph, paths[end_node], weight="weight"))
-                    for end_node in end_nodes if end_node in paths]
-    if not path_weights:
+    spectrum_data = spectrum.velocity_spectrum.max() - spectrum.velocity_spectrum
+    spectrum_times = np.asarray(spectrum.times, dtype=np.float32)
+    spectrum_velocities = np.asarray(spectrum.velocities, dtype=np.float32)
+
+    # Calculate times of graph nodes
+    times_step_samples = int(times_step // spectrum.sample_interval)
+    node_times_ix = np.arange(0, len(spectrum_times), times_step_samples)
+    node_times_ix[-1] = len(spectrum_times) - 1
+    node_times = spectrum_times[node_times_ix]
+
+    # Estimate velocity bounds for each time where nodes are placed
+    if bounds is not None:
+        min_vel_bound = bounds[0](node_times)
+        max_vel_bound = bounds[1](node_times)
+    else:
+        center_vel = init(node_times)
+        min_vel_bound = center_vel * (1 - relative_margin)
+        max_vel_bound = center_vel * (1 + relative_margin)
+    min_spectrum_velocity = spectrum_velocities.min()
+    max_spectrum_velocity = spectrum_velocities.max()
+    min_vel_bound = np.clip(min_vel_bound, min_spectrum_velocity, max_spectrum_velocity)
+    max_vel_bound = np.clip(max_vel_bound, min_spectrum_velocity, max_spectrum_velocity)
+    min_vel_bound = np.minimum(min_vel_bound, max_vel_bound)
+    max_vel_bound = np.maximum(min_vel_bound, max_vel_bound)
+
+    # Estimate node velocities for each time
+    final_hodograph_time_max = np.sqrt(node_times**2 + (max_offset * 1000 / min_vel_bound)**2)  # ms
+    final_hodograph_time_min = np.sqrt(node_times**2 + (max_offset * 1000 / max_vel_bound)**2)  # ms
```

Comment:
please unify `max`/`min` notation for this variables
```suggestion
    max_correction_time = np.sqrt(node_times**2 + (max_offset * 1000 / min_vel_bound)**2)  # ms
    min_correction_time = np.sqrt(node_times**2 + (max_offset * 1000 / max_vel_bound)**2)  # ms
```

Reply:
Done